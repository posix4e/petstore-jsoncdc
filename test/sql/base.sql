\set ECHO none
SET client_min_messages TO error;
\t on
\x off
SELECT 'created logical replication slot'
  FROM pg_create_logical_replication_slot('jsoncdc', 'jsoncdc');

BEGIN;

CREATE SCHEMA IF NOT EXISTS jsoncdc;
SET LOCAL search_path TO jsoncdc, public;
CREATE EXTENSION IF NOT EXISTS hstore;

CREATE TABLE IF NOT EXISTS test (
  i   integer PRIMARY KEY,
  h   hstore NOT NULL DEFAULT ''
);

INSERT INTO test (i) SELECT i FROM generate_series(1, 8) AS i;

CREATE VIEW changedata AS
SELECT data FROM pg_logical_slot_get_changes('jsoncdc', NULL, NULL);

END;

BEGIN;
SET LOCAL search_path TO jsoncdc, public;
UPDATE test SET i = i * 10 WHERE i % 3 = 0;
UPDATE test SET h = hstore('i', i::text)||hstore('2i', (2*i)::text);
DELETE FROM test WHERE i % 2 = 1;
END;

--- Displays only generated JSON, no replication slot metadata, and omits
--- transaction IDs (which would confuse the testing framework because they
--- vary from run to run).
SELECT * FROM jsoncdc.changedata
 WHERE NOT (data LIKE '{ "begin": %' OR data LIKE '{ "commit": %');

SELECT 'deleted logical replication slot'
  FROM pg_drop_replication_slot('jsoncdc');

DROP SCHEMA jsoncdc CASCADE;
