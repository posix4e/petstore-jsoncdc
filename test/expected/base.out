\set ECHO none
 created logical replication slot

 { "table": "jsoncdc.test", "schema": [{"i":"integer"},{"h":"jsoncdc.hstore"}] }
 { "insert": {"i":1,"h":{}} }
 { "insert": {"i":2,"h":{}} }
 { "insert": {"i":3,"h":{}} }
 { "insert": {"i":4,"h":{}} }
 { "insert": {"i":5,"h":{}} }
 { "insert": {"i":6,"h":{}} }
 { "insert": {"i":7,"h":{}} }
 { "insert": {"i":8,"h":{}} }
 { "table": "jsoncdc.test", "schema": [{"i":"integer"},{"h":"jsoncdc.hstore"}] }
 { "update": {"i":30,"h":{}}, "@": {"i":3,"h":null} }
 { "update": {"i":60,"h":{}}, "@": {"i":6,"h":null} }
 { "update": {"i":1,"h":{"i": "1", "2i": "2"}} }
 { "update": {"i":2,"h":{"i": "2", "2i": "4"}} }
 { "update": {"i":4,"h":{"i": "4", "2i": "8"}} }
 { "update": {"i":5,"h":{"i": "5", "2i": "10"}} }
 { "update": {"i":7,"h":{"i": "7", "2i": "14"}} }
 { "update": {"i":8,"h":{"i": "8", "2i": "16"}} }
 { "update": {"i":30,"h":{"i": "30", "2i": "60"}} }
 { "update": {"i":60,"h":{"i": "60", "2i": "120"}} }
 { "delete": {}, "@": {"i":1,"h":null} }
 { "delete": {}, "@": {"i":5,"h":null} }
 { "delete": {}, "@": {"i":7,"h":null} }

 deleted logical replication slot

